Sources/Private/Experimental/ExperimentalAnimationLayer.swift

```diff
@@ -220,12 +223,17 @@ extension ExperimentalAnimationLayer: RootAnimationLayer {
     // Unimplemented / unused
   }
 
-  func setValueProvider(_: AnyValueProvider, keypath _: AnimationKeypath) {
-    LottieLogger.shared.assertionFailure("`AnimationKeypath`s are currently unsupported")
+  func setValueProvider(_ valueProvider: AnyValueProvider, keypath: AnimationKeypath) {
+    // TODO: We need to rebuild the current animation after registering a value provider,
+    // since any existing `CAAnimation`s could now be out of date.
+    valueProviderStore.setValueProvider(valueProvider, keypath: keypath)
```

Comment:
These changes currently don't kick in until the next time an animation is setup, which is incorrect. Will work on a fix for this in a follow-up.

<img src="https://user-images.githubusercontent.com/1811727/149424992-b35df05e-93eb-470b-bb4c-2a734bcb9ee2.gif" width=350>

Reply:
Fixed in #1483